BLUESPAWN-client/src/util/configurations/RegistryKey.cpp

```diff
@@ -0,0 +1,485 @@
+
+			while(*data){
+				std::wstring str = data;
+				strings.emplace_back(data);
+				wsLogString += str;
+
+				data += str.length() + 1;
+			}
+
+			return strings;
+		}
+
+		return std::nullopt;
+	}
+
+	bool RegistryKey::SetRawValue(const std::wstring& name, AllocationWrapper bytes, DWORD dwType) const {
+		if(!Exists()){
+			return false;
+		}
+
+		LSTATUS status = RegSetValueEx(hkBackingKey, name.c_str(), 0, dwType, reinterpret_cast<BYTE*>((LPVOID) bytes), bytes.GetSize());
+		if(status != ERROR_SUCCESS){
+			SetLastError(status);
+			return false;
+		}
+
+		return true;
+	}
+
+	template<class T>
+	bool RegistryKey::SetValue(const std::wstring& name, T value, DWORD size, DWORD type) const {
+		return SetRawValue(name, { reinterpret_cast<BYTE*>(value), size, AllocationWrapper::STACK_ALLOC }, type);
+	}
+
+	template<>
+	bool RegistryKey::SetValue(const std::wstring& name, LPCWSTR value, DWORD size, DWORD type) const {
+		return RegistryKey::SetRawValue(name, { PBYTE(value), wcslen(value), AllocationWrapper::STACK_ALLOC }, type);
+	}
+	template bool RegistryKey::SetValue<LPCWSTR>(const std::wstring& name, LPCWSTR value, DWORD size, DWORD type) const;
+
+	template<>
+	bool RegistryKey::SetValue(const std::wstring& name, LPCSTR value, DWORD size, DWORD type) const {
+		return RegistryKey::SetRawValue(name, { PBYTE(value), strlen(value), AllocationWrapper::STACK_ALLOC }, type);
+	}
+	template bool RegistryKey::SetValue<LPCSTR>(const std::wstring& name, LPCSTR value, DWORD size, DWORD type) const;
+
+	template<>
+	bool RegistryKey::SetValue(const std::wstring& name, DWORD value, DWORD size, DWORD type) const {
+		return SetRawValue(name, { reinterpret_cast<BYTE*>(&value), 4, AllocationWrapper::STACK_ALLOC }, REG_DWORD);
+	}
+	template bool RegistryKey::SetValue<DWORD>(const std::wstring& name, DWORD value, DWORD size, DWORD type) const;
+
+	template<> 
+	bool RegistryKey::SetValue(const std::wstring& name, std::wstring value, DWORD size, DWORD type) const {
+		return SetValue<LPCWSTR>(name, value.c_str(), static_cast<DWORD>((value.size() + 1) * 2), REG_SZ);
+	}
+	template bool RegistryKey::SetValue<std::wstring>(const std::wstring& name, std::wstring value, DWORD size, DWORD type) const;
+
+	template<>
+	bool RegistryKey::SetValue(const std::wstring& name, std::string value, DWORD size, DWORD type) const {
+		return SetValue<std::wstring>(name, StringToWidestring(value));
+	}
+	template bool RegistryKey::SetValue<std::string>(const std::wstring& name, std::string value, DWORD size, DWORD type) const;
+
+	template<>
+	bool RegistryKey::SetValue(const std::wstring& name, std::vector<std::wstring> value, DWORD _size, DWORD type) const {
+		SIZE_T size = 1;
+		for(auto string : value){
+			size += (string.length() + 1);
+		}
+
+		WCHAR* data = new WCHAR[size];
+		unsigned ptr = 0;
+
+		std::wstring wsLogStatement{};
+		for(auto string : value){
+			wsLogStatement += string + L"; ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'wsLogStatement' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4967854)